File: 513. Find Bottom Left Tree Value/findBottomLeftValue.py
from collections import deque
from typing import Optional
# Definition for a binary tree node.
class TreeNode:
  def __init__(self, val=0, left=None, right=None):
    self.val = val
    self.left = left
    self.right = right

class Solution:
  def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
    queue = deque([root])

    while queue:
      node = queue.popleft()

      if node.right:
        queue.append(node.right)

      if node.left:
        queue.append(node.left)

    return node.val

root = TreeNode(1)
root.left = TreeNode(2)
root.right = TreeNode(3)

root.left.left = TreeNode(4)

root.right.left = TreeNode(5)
root.right.right = TreeNode(6)

root.right.left.left = TreeNode(7)

solution = Solution()
print('findBottomLeftValue: ', solution.findBottomLeftValue(root))

File: 513. Find Bottom Left Tree Value/findBottomLeftValue.js

// Definition for a binary tree node.
class TreeNode {
  constructor(val = 0, left = null, right = null) {
    this.val = val;
    this.left = left;
    this.right = right;
  }
}

/**
 * @param {TreeNode} root
 * @return {number}
 */
const findBottomLeftValue = (root) => {
  const queue = [root];
  let node = null;

  while(queue.length) {
    node = queue.shift()

    if (node.right) {
      queue.push(node.right);
    }

    if (node.left) {
      queue.push(node.left);
    }
  }

  return node.val; 
};

const root = new TreeNode(1);
root.left = new TreeNode(2);
root.right = new TreeNode(3);

root.left.left = new TreeNode(4);

root.right.left = new TreeNode(5);
root.right.right = new TreeNode(6);

root.right.left.left = new TreeNode(7);

console.log('findBottomLeftValue: ', findBottomLeftValue(root));